src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java

```diff
@@ -0,0 +1,231 @@
+                                         UserService userService,
+                                         LocationReferenceDataService locationRefDataService) {
+
+        this.objectMapper = objectMapper;
+        this.courtLocationUtils = courtLocationUtils;
+        this.locationReferenceDataService = locationReferenceDataService;
+        this.stateFlowEngine = stateFlowEngine;
+        this.coreCaseUserService = coreCaseUserService;
+        this.userService = userService;
+        this.locationRefDataService = locationRefDataService;
+    }
+
+    @Override
+    public CallbackResponse execute(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        LocalDateTime dateTime = LocalDateTime.now();
+
+        CallbackResponse errorResponse = checkSubmitError(callbackParams, caseData, dateTime);
+        if (errorResponse != null) {
+            return errorResponse;
+        }
+
+        var isRespondent1 = findOutIsRespondent1(callbackParams);
+
+        List<LocationRefData> locations = fetchLocationData(callbackParams);
+
+        RequestedCourt requestedCourt1 = createRequestedCourt(locations, caseData);
+
+        CaseData.CaseDataBuilder updatedCaseData = updateCaseData(caseData, isRespondent1, requestedCourt1);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.build().toMap(objectMapper))
+            .build();
+    }
+
+    private CallbackResponse checkSubmitError(CallbackParams callbackParams,
+                                              CaseData caseData,
+                                              LocalDateTime dateTime) {
+        // Show error message if defendant tries to submit response again
+        if (hasDefendantAlreadySubmitted(callbackParams, caseData)) {
+            return createErrorResponse(ERROR_DEFENDANT_RESPONSE_SUBMITTED);
+        }
+
+        // Show error message if defendant tries to submit a response after deadline has passed
+        if (isResponseSubmissionPassedDeadline(callbackParams, caseData, dateTime)) {
+            return createErrorResponse("You cannot submit a response now as you have passed your deadline");
+        }
+
+        return null;
+    }
+
+    private YesOrNo findOutIsRespondent1(CallbackParams callbackParams) {
+        YesOrNo isRespondent1 = YES;
+        if (solicitorRepresentsOnlyOneOrBothRespondents(callbackParams, RESPONDENTSOLICITORTWO)) {
+            // 1V2 Different Solicitors + Respondent 2 only
+            isRespondent1 = NO;
+        }
+        return isRespondent1;
+    }
+
+    private CallbackResponse createErrorResponse(String errorMessage) {
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .errors(List.of(errorMessage))
+            .build();
+    }
+
+    private CaseData.CaseDataBuilder updateCaseData(CaseData caseData, YesOrNo isRespondent1, RequestedCourt requestedCourt1) {
+        var updatedCaseData = caseData.toBuilder()
```

Comment:
same here with var, and you are creating a new caseData object, you will override the data saved when I believe, what you are trying to do is just update some values, so you shouldn't create a new instance with the toBuilder()